### PR TITLE
Added GH action for building the docs

### PR DESCRIPTION
**Pull request by Steveb-p**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 1.13 as base

Creates a GH action that fails if `mkdocs` binary that is used for building the docs has any warnings.

For example in this particular case there are missing links.

### CODE REVIEW

File: .github/workflows/build.yaml
Comment (ibexa-yuna) [inline .github/workflows/build.yaml:14]: Why matrix?

Comment (Steveb-p) [reply .github/workflows/build.yaml:14]: No particular reason.

I've used a Python template where there were multiple versions and decided that it wasn't worth it to remove it, in case we need to build these against multiple versions.

**Comment by DominikaK** [comment]
@Steveb-p: If a PR fails this check, will it still be possible to merge despite that (in cases where, for example, we handle some redirects on the side of our hosting)? Or will it block merging?

**Comment by Steveb-p** [comment]
> @Steveb-p: If a PR fails this check, will it still be possible to merge despite that (in cases where, for example, we handle some redirects on the side of our hosting)? Or will it block merging?

This depends on the repository settings afaik. In case of this PR you can see that "Merge pull request" button is still there, even if it's not "green" - indicating that you can merge it, but checks aren't passing.

**Comment by DominikaK** [comment]
Before merging this we'll need to clean up some preexisting broken links, mostly in release notes.

But great thanks for the contrib anyway ❤️

File: README.md
Comment (DominikaK) [inline README.md:24]:
```suggestion
To build and preview your changes locally, you will need Python along with its package manager (`pip`).
```

File: README.md
Comment (DominikaK) [inline README.md:25]:
```suggestion
Other required tools will be installed by `pip` using the following command:
```

File: README.md
Comment (DominikaK) [inline README.md:30]:
```suggestion
Then you can run:
```

**Review by DominikaK — APPROVED** [review]
Most pre-existing link issues have been fixed in the meantime, so this should be green now.

File: README.md
Comment (Steveb-p) [reply README.md:24]: I was wrong all my life...

https://www.faa.gov/about/initiatives/plain_language/articles/its/#:~:text=It's%20is%20a%20contraction%2C%20meaning,%3A%20Its%20color%20is%20red.)

:open_mouth:

**Comment by Steveb-p** [comment]
> Most pre-existing link issues have been fixed in the meantime, so this should be green now.

After rebase there are still some broken links reported during build :crying_cat_face: 

```
WARNING -  Documentation file 'api/field_type_reference.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'api/field_type_search.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'api/js_client.md' contains a link to 'guide/search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'api/public_php_api_search.md' contains a link to 'guide/search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'api/public_php_api_search.md' contains a link to 'guide/search/sort_clause/reference.md' which is not found in the documentation files. 
WARNING -  Documentation file 'api/public_php_api_search.md' contains a link to 'guide/search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/clustering.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/content_model.md' contains a link to 'guide/search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/controllers.md' contains a link to 'guide/search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/landing_page_rendering.md' contains a link to 'guide/best_practices.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/other_extensibility.md' contains a link to 'cookbook/creating_landing_page_blocks_(enterprise).md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/performance.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/performance.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/performance.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/persistence_cache.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/search/search.md' contains a link to 'guide/api/public_php_api_search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/search/search.md' contains a link to 'guide/api/public_php_api_search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/search/search.md' contains a link to 'guide/api/public_php_api_search.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/search/solr.md' contains a link to 'guide/search/architecture.md' which is not found in the documentation files. 
WARNING -  Documentation file 'guide/search/solr.md' contains a link to 'guide/api/field_type_reference.md' which is not found in the documentation files. 
WARNING -  Documentation file 'migrating/migrating_from_ez_publish.md' contains a link to 'tutorials/extending_admin_ui/extending_admin_ui.md' which is not found in the documentation files. 
WARNING -  Documentation file 'migrating/migrating_from_ez_publish.md' contains a link to 'guide/page_rendering.md' which is not found in the documentation files. 
WARNING -  Documentation file 'releases/ez_platform_v1.11.0.md' contains a link to 'releases/img\relation_single_allowed_cts.png' which is not found in the documentation files. 
WARNING -  Documentation file 'releases/ez_platform_v1.7.0_lts.md' contains a link to 'guide/solr.md' which is not found in the documentation files. 
WARNING -  Documentation file 'releases/ez_platform_v3.0.md' contains a link to 'guidelines/components/tooltips.md' which is not found in the documentation files. 
WARNING -  Documentation file 'releases/ez_platform_v3.0.md' contains a link to 'extending/extending_tooltips.md' which is not found in the documentation files. 
WARNING -  Documentation file 'tutorials/enterprise_beginner/2_prepare_the_landing_page.md' contains a link to 'guide/page_rendering.md' which is not found in the documentation files. 
WARNING -  Documentation file 'tutorials/extending_platformui/2_set_up_the_configuration.md' contains a link to 'tutorials/extending_platformui/siteaccess.md' which is not found in the documentation files. 
WARNING -  Documentation file 'tutorials/extending_platformui/conclusion.md' contains a link to 'api/public_php_api_customization.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/4_6_upgrade_workflow.md' contains a link to 'guide/extending/extending_workflow.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/4_7_upgrade_extensions.md' contains a link to 'guide/extending/extending_udw.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/4_7_upgrade_extensions.md' contains a link to 'guide/extending/extending_tabs.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/4_upgrade_the_code.md' contains a link to 'guide/project_organization.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/5_update_2.2.md' contains a link to 'guide/extending/extending_page.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/5_update_2.2.md' contains a link to 'guide/extending/extending_page.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/5_update_2.2.md' contains a link to 'guide/extending/extending_page.md' which is not found in the documentation files. 
WARNING -  Documentation file 'updating/5_update_2.5.md' contains a link to 'guide/databases.md' which is not found in the documentation files. 
```

**Comment by DominikaK** [comment]
Ahh, 1.13 💓

**Comment by Steveb-p** [comment]
> Ahh, 1.13

Yup, and that's the whole point - each and every version has to be complete without broken links :wink:

**Comment by DominikaK** [comment]
And **now** the links should be fixed for all branches and this is ready for rebase.

**Comment by Steveb-p** [comment]
Removed the build upload, as it doesn't make sense if we're using a commercial deploy to readthedocs.com.

**Review by ibexa-yuna — APPROVED** [review]
Workflow: LGTM

**Comment by Steveb-p** [comment]
@DominikaK I see there are conflicts when trying to merge it up to `2.5` and above. Can we resolve them tomorrow morning?

**Comment by DominikaK** [comment]
@Steveb-p Let me handle other branches, it's doc, so we do things differently ;)

**Pull request merged**
